lib/miq_automation_engine/service_models/miq_ae_service_container_deployment.rb

```diff
@@ -0,0 +1,255 @@
+    expose :deployed_ems, :association => true
+    expose :deployed_on_ems, :association => true
+    expose :automation_task, :association => true
+    expose :roles_addresses
+    expose :container_nodes_by_role
+    expose :ssh_auth
+    expose :rhsm_auth
+    expose :generate_ansible_yaml
+    expose :generate_ansible_inventory_for_subscription
+    expose :add_deployment_provider
+    SUBSCRIPTION_REPOS = ["rhel-7-server-rh-common-rpms", "rhel-7-server-rpms", "rhel-7-server-extras-rpms", "rhel-7-server-ose-3.2-rpms"].freeze
+    ANSIBLE_LOG = "/tmp/ansible.log".freeze
+    ANSIBLE_ERROR_LOG = "/tmp/openshift-ansible.log".freeze
+    INVENTORIES_PATH = "/usr/share/ansible/openshift-ansible/".freeze
+    SSH_AGENT_PATH = "/tmp/ssh_manageiq/".freeze
+
+    def subscribe_deployment_master(deployment_master, user)
+      rhsub_user = rhsm_auth.userid
+      rhsub_pass = rhsm_auth.password
+      rhsm_sku = rhsm_auth.rhsm_sku
+      perform_agent_commands(deployment_master, user, ["sudo subscription-manager register --username='#{rhsub_user}' --password='#{rhsub_pass}'"])
+      pool_cmdret = perform_agent_commands(deployment_master, user, ["sudo subscription-manager list --available --matches='#{rhsm_sku}' --pool-only"])
+      pool_id = pool_cmdret[:stdout].split("\n").first.to_s.strip
+      perform_agent_commands(deployment_master, user, ["sudo subscription-manager attach --pool='#{pool_id}'"]) unless pool_id.blank?
+      enabled_repos_cmdret = perform_agent_commands(deployment_master, user, ['sudo subscription-manager repos --disable="*"',
+                                                                              "sudo subscription-manager repos --enable=#{SUBSCRIPTION_REPOS.join(" --enable=")}",
+                                                                              "sudo subscription-manager repos --list-enabled"])
+      enabled_repos = enabled_repos_cmdret[:stdout].split(" ").to_a.map(&:strip).reject(&:empty?)
+      missing_repos = SUBSCRIPTION_REPOS - enabled_repos
+      raise StandardError, "couldn't register to following repos : #{missing_repos.join(',')} with SKU : #{rhsm_sku}" unless missing_repos.empty?
+    end
+
+    def subscribe_cluster(deployment_master, user)
+      perform_agent_commands(deployment_master, user, ["sudo mv ~/rhel_subscribe_inventory.yaml #{INVENTORIES_PATH}"])
+      run_playbook_command(deployment_master, user, "ansible-playbook /usr/share/ansible/openshift-ansible/playbooks/byo/rhel_subscribe.yml -i "\
+                         "#{INVENTORIES_PATH}rhel_subscribe_inventory.yaml 1> #{ANSIBLE_LOG} 2> #{ANSIBLE_ERROR_LOG} < /dev/null")
+    end
+
+    def provision_tagged_tasks
+      tasks = []
+      MiqRequest.all.each do |prov|
+        next unless prov.get_tags[:miq_openshift_deploy] &&
+                    (prov.get_tags[:miq_openshift_deploy].include?(get_custom_tag("master")) ||
+                      prov.get_tags[:miq_openshift_deploy].include?(get_custom_tag("node")))
```

Comment:
@kbrock Can you help out here.  Looping over every request and check for specific tags is not scalable.